drivers/power/lt7170/iio_lt7170.c

```diff
@@ -0,0 +1,1039 @@
+				       &status_byte);
+		if (ret)
+			return ret;
+
+		val = (int32_t)status_byte;
+
+		return iio_format_value(buf, len, IIO_VAL_INT, 1, &val);
+	}
+}
+
+/**
+ * @brief Initializes the LT7170 IIO descriptor.
+ * @param iio_desc - The iio device descriptor.
+ * @param init_param - The structure that contains the device initial parameters.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int lt7170_iio_init(struct lt7170_iio_desc **iio_desc,
+		    struct lt7170_iio_desc_init_param *init_param)
+{
+	struct lt7170_iio_desc *descriptor;
+	int ret;
+
+	if (!init_param || !init_param->lt7170_init_param)
+		return -EINVAL;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = lt7170_init(&descriptor->lt7170_dev,
+			  init_param->lt7170_init_param);
+	if (ret)
+		goto dev_err;
+
+	descriptor->iio_dev = &lt7170_iio_dev;
+
+	*iio_desc = descriptor;
+
+	return 0;
+
+dev_err:
+	no_os_free(descriptor);
+
+	return ret;
+}
+
+/**
+ * @brief Free resources allocated by the init function.
+ * @param iio_desc - The iio device descriptor.
+ * @return 0 in case of success, an error code otherwise.
+ */
+int lt7170_iio_remove(struct lt7170_iio_desc *iio_desc)
+{
+	if (!iio_desc)
+		return -ENODEV;
+
+	no_os_free(iio_desc->iio_dev->channels);
+	lt7170_remove(iio_desc->lt7170_dev);
+	no_os_free(iio_desc);
+
+	return 0;
+}
+
+static struct iio_attribute lt7170_input_attrs[] = {
+	{
+		.name = "raw",
+		.show = lt7170_iio_read_raw
+	},
+	{
+		.name = "scale",
+		.show = lt7170_iio_read_scale
+	},
+	END_ATTRIBUTES_ARRAY
+};
+
+static struct iio_attribute lt7170_output_attrs[] = {
+	{
+		.name = "raw",
+		.show = lt7170_iio_read_raw
+	},
+	{
+		.name = "scale",
+		.show = lt7170_iio_read_scale
+	},
+	{
+		.name = "enable",
+		.show = lt7170_iio_read_enable,
+		.store = lt7170_iio_write_enable,
+		.priv = LT7170_IIO_OUTPUT_ENABLE,
+	},
+	{
+		.name = "enable_available",
+		.show = lt7170_iio_read_enable_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	{
+		.name = "vout_command",
+		.show = lt7170_iio_read_vout,
+		.store = lt7170_iio_write_vout,
+		.priv = LT7170_VOUT_COMMAND
+	},
+	{
+		.name = "vout_max",
+		.show = lt7170_iio_read_vout,
+		.store = lt7170_iio_write_vout,
+		.priv = LT7170_VOUT_MAX
+	},
+	{
+		.name = "vout_margin_high",
+		.show = lt7170_iio_read_vout,
+		.store = lt7170_iio_write_vout,
+		.priv = LT7170_VOUT_MARGIN_HIGH
+	},
+	{
+		.name = "vout_margin_low",
+		.show = lt7170_iio_read_vout,
+		.store = lt7170_iio_write_vout,
+		.priv = LT7170_VOUT_MARGIN_LOW
+	},
+	{
+		.name = "phase",
+		.show = lt7170_iio_read_pwm,
+		.store = lt7170_iio_write_pwm,
+		.priv = LT7170_IIO_PWM_PHASE
+	},
+	{
+		.name = "pulse_skipping",
+		.show = lt7170_iio_read_enable,
+		.store = lt7170_iio_write_enable,
+		.priv = LT7170_IIO_PULSE_ENABLE
+	},
+	{
+		.name = "pulse_skipping_available",
+		.show = lt7170_iio_read_enable_available,
+		.shared = IIO_SHARED_BY_ALL
+	},
+	END_ATTRIBUTES_ARRAY
```

Comment:
Most of these are not IIO attributes defined in the IIO ABI (https://www.kernel.org/doc/Documentation/ABI/testing/sysfs-bus-iio). Thus, they should be debug attributes instead.